module-4/hydra/hydra.py

```diff
@@ -1 +1,82 @@
-print("Initializing Hydra")
\ No newline at end of file
+import argparse
+import paramiko
+import itertools
+import string
+import time
+import signal
+import sys
+
+def signal_handler(sig, frame):
+    print('\n[!] Signal received. Exiting gracefully.')
+    sys.exit(0)
+
+signal.signal(signal.SIGINT, signal_handler)  # Handle Ctrl+C (SIGINT)
+
+def brute_force_ssh(username, server, charset, min_length, max_length):
```

Comment:
add type [hints](https://peps.python.org/pep-0484/) and a docstring (see [docs](https://peps.python.org/pep-0257/)) to your function definition

Reply:
type hints were meant like follows:

```python
def brute_force_ssh(
    username: str,
    server: str,
    charset: str,
    min_length: int,
    max_length: int):
    # your implementation goes here
```